pkg/resource/bucket/hook.go

```diff
@@ -0,0 +1,171 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package bucket
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/s3"
+)
+
+func (rm *resourceManager) syncPutFields(
+	ctx context.Context,
+	r *resource,
+) (synced *resource, err error) {
+	// Copy resource for later patching
+	synced = &resource{r.ko.DeepCopy()}
+
+	diff, err := rm.diffLogging(ctx, synced)
```

Comment:
instead of doing this, see my comment below about adding the `diffLogging` call in the `delta_pre_compare` hook and passing the `delta` object to this function. That way, here you can simply do:

```go
if delta.DifferentAt("Spec.Logging")
```